pkg/operator/controllers/subnets/subnet_nsg.go

```diff
@@ -30,14 +30,18 @@ func (r *reconcileManager) ensureSubnetNSG(ctx context.Context, s subnet.Subnet)
 		return err
 	}
 
-	if !strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
-		r.log.Infof("Fixing NSG from %s to %s", *subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID)
+	// if the NSG is assigned && it's the correct NSG - do nothing
+	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup != nil && strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
+		return nil
+	}
 
-		subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
-		err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
-		if err != nil {
-			return err
-		}
+	// else the NSG assignment needs to be corrected
+	oldNSG := "nil"
+	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
```

Comment:
Do we need an `oldNSG` var, or does something like this work?
```suggestion
	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
		r.log.Infof("Fixing NSG from '%s' to '%s'", subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID)
		subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
		
		err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
		if err != nil {
			return err
		}
	}
```
What you have now makes the log message more clear, but I think an empty string with single quotes would get the point across.

Reply:
alternatively use `%q` instead of `'%s'`.

Reply:
you could alternatively use `%q` instead of `'%s'`.